docs/datapack_spec.md

```diff
@@ -0,0 +1,173 @@
+
+* `content`: The content of the resource that complies with the content schema defined in the corresponding schemapack.
+
+    ```yaml
+
+    content: # <- content schema embedded here
+        name: Example Resource
+        description: This is an example resource.
+    ```
+
+* `relations`: A dictionary containing the relations of the resource to other resources. Each key is the name of a relation mapped to their datapack relation definitions. Each value contains the target class and target resource(s) of the relation.
+
+  Relation definition must follow the following structure:
+
+    * `targetClass`:  The name of the referenced class.
+
+    * `targetResources`: The ID(s) of target resources of the targetClass. Based on the schemapack relation definition, this field can get the following values:
+
+
+    |                             | **mandatory.target = True** | **mandatory.target = False**        |
+    | --------------------------- | --------------------------- | ----------------------------------- |
+    | **multiple.target = True**  | set of resource IDs         | set of resource IDs or an empty set |
+    | **multiple.target = False** | a single ID                 | a single ID or None                 |
+
+    ```yaml
+
+    relations:
+        samples: # <- relation name
+            targetClass: Sample
+            targetResources: [sample1, sample2]
+    ```
+
+
+A full datapack example:
+
+```yaml
+datapack: 3.0.0
+resources:
+    Experiment:
+        exp1:
+            content:
+                name: Experiment 1
+                description: This is the first experiment.
+            relations:
+                samples:
+                    targetClass: Sample
+                    targetResources: sample1
+        exp2:
+            content:
+                name: Experiment 2
+                description: This is the second experiment.
+            relations:
+                samples:
+                    targetClass: Sample
+                    targetResources: sample1
+    Sample:
+        sample1:
+            content:
+                name: Sample 1
+                description: This is the first sample.
+```
+
+
+`rootResource` *(optional)*: Defines the id of the resource of the class defined in `className` that should act as root. This means that, in addition to the root resource itself, the datapack must only contain resources that are direct or indirect (dependencies of dependencies) of the root resource.
+
+`rootClass` *(optional)*: Defines the class name of the resource that should act as root.
+
+<div style="border: 1px solid #007acc; border-left: 4px solid #007acc; padding: 1em; border-radius: 6px; background: transparent; margin-bottom: 1em;">
+  <strong>Note:</strong> A rooted <code>datapack</code> requires its corresponding <code>schemapack</code> to be rooted as well. If you're validating a rooted <code>datapack</code> agains a schema, make sure that its <code>schemapack</code> counterpart is also rooted.
```

Comment:
This looks weird in the rich diff, every `code` block seems to induce a new line